code/modules/atmospherics/pipe_network.dm

```diff
@@ -1,49 +1,48 @@
 var/global/list/datum/pipe_network/pipe_networks = list()
 
 /datum/pipe_network
-	var/list/datum/gas_mixture/gases = list() //All of the gas_mixtures continuously connected in this network
+	var/list/datum/gas_mixture/gases //All of the gas_mixtures continuously connected in this network
 
-	var/list/obj/machinery/atmospherics/normal_members = list()
-	var/list/datum/pipeline/line_members = list()
-		//membership roster to go through for updates and what not
+	var/list/obj/machinery/atmospherics/normal_members
+	var/list/datum/pipeline/line_members
+	//membership roster to go through for updates and what not
 
 	var/update = 1
-	var/datum/gas_mixture/air_transient = null
+	var/datum/gas_mixture/air_transient
 
 /datum/pipe_network/New()
-	air_transient = new /datum/gas_mixture
-
 	..()
+	air_transient = new /datum/gas_mixture
+	src.gases = list()
+	src.normal_members = list()
+	src.line_members = list()
 
 /datum/pipe_network/disposing()
 	update = 0
 	pipe_networks -= src
-	if (gases)
-		gases.len = 0
+
 	gases = null
-	if (normal_members)
-		for(var/obj/machinery/atmospherics/machine in normal_members)
-			machine.network_disposing(src)
-		normal_members.len = 0
-	normal_members = 0
-	if (line_members)
-		for(var/datum/pipeline/member in line_members)
-			member.network = null
-		line_members.len = 0
+
+	for(var/obj/machinery/atmospherics/machine as anything in normal_members)
```

Comment:
what happens when one of them blows up? you'll get a null in this list

Reply:
Tested, the exploding normal_member calls dispose on the pipe_network while still existing during disposing(), meaning there will never be a null in this list, hopefully.

Reply:
What about any of the other lists you are skipping the typechecking for?

Reply:
air_disposing_hook() is used by normal members for removing gases from /datum/pipe_network/var/list/gases, pipeline disposing removes its gases and itself from the pipe_network, and pipelines are disposed whenever a member disposes.